pkg/runctl/run_manager_test.go

```diff
@@ -2111,3 +2220,101 @@ func Test__runManager__getTimeout__retrievesTheDefaultPipelineTimeoutIfTimeoutIs
 	//VERIFY
 	assert.DeepEqual(t, defaultTimeout, result)
 }
+
+func Test__runManager_GetRun_Missing(t *testing.T) {
+	t.Parallel()
+
+	// SETUP
+	h := newTestHelper1(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockFactory, mockPipelineRun, mockSecretProvider := h.prepareMocks(mockCtrl)
+	h.addTektonTaskRun(mockFactory)
+	examinee := newRunManager(mockFactory, mockSecretProvider)
+
+	// EXERCISE
+	run, resultError := examinee.GetRun(h.ctx, mockPipelineRun)
+
+	// VERIFY
+	assert.NilError(t, resultError)
+	assert.Assert(t, run != nil)
+
+	mockPipelineRun.EXPECT().UpdateState(gomock.Any(), gomock.Any()).Times(0)
+}
+
+func Test__runManager_GetRun_Existing(t *testing.T) {
+	t.Parallel()
+
+	// SETUP
+	h := newTestHelper1(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockFactory, mockPipelineRun, mockSecretProvider := h.prepareMocks(mockCtrl)
+	examinee := newRunManager(mockFactory, mockSecretProvider)
+
+	// EXERCISE
+	run, resultError := examinee.GetRun(h.ctx, mockPipelineRun)
+
+	// VERIFY
+	assert.NilError(t, resultError)
+	assert.Assert(t, run == nil)
+
+	mockPipelineRun.EXPECT().UpdateState(gomock.Any(), gomock.Any()).Times(0)
+}
+
+func Test__runManager_DeleteRun_Success(t *testing.T) {
+	t.Parallel()
+
+	// SETUP
+	h := newTestHelper1(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockFactory, mockPipelineRun, mockSecretProvider := h.prepareMocks(mockCtrl)
+	h.addTektonTaskRun(mockFactory)
+	examinee := newRunManager(mockFactory, mockSecretProvider)
+
+	// EXERCISE
+	resultError := examinee.DeleteRun(h.ctx, mockPipelineRun)
+
+	// VERIFY
+	assert.NilError(t, resultError)
+
+	mockPipelineRun.EXPECT().UpdateState(gomock.Any(), gomock.Any()).Times(0)
+}
+
+func Test__runManager_DeleteRun_Missing(t *testing.T) {
+	t.Parallel()
+
+	// SETUP
+	h := newTestHelper1(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockFactory, mockPipelineRun, mockSecretProvider := h.prepareMocks(mockCtrl)
+	examinee := newRunManager(mockFactory, mockSecretProvider)
+
+	// EXERCISE
+	resultError := examinee.DeleteRun(h.ctx, mockPipelineRun)
+
+	// VERIFY
+	assert.NilError(t, resultError)
+	mockPipelineRun.EXPECT().UpdateState(gomock.Any(), gomock.Any()).Times(0)
+}
+
+func Test__runManager_DeleteRun_MissingRunNamespace(t *testing.T) {
+	t.Parallel()
+
+	// SETUP
+	h := newTestHelperNoRunNamespace(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockFactory, mockPipelineRun, mockSecretProvider := h.prepareMocks(mockCtrl)
+	examinee := newRunManager(mockFactory, mockSecretProvider)
+	mockPipelineRun.EXPECT().GetName().Return("foo").Times(1)
+	// EXERCISE
+	resultError := examinee.DeleteRun(h.ctx, mockPipelineRun)
+
+	// VERIFY
+	assert.Error(t, resultError, `cannot delete taskrun, runnamespace not set in "foo"`)
+
+	mockPipelineRun.EXPECT().UpdateState(gomock.Any(), gomock.Any()).Times(0)
+}
```

Comment:
Add a test showing that certain errors are flagged as retry-able.

Reply:
See https://github.com/SAP/stewardci-core/pull/345/commits/130f3e84ba3eeb42b9c29a6e8c0ed0b3a838c8e8